packages/with-zod/src/with-zod.test.ts

```diff
@@ -64,4 +64,26 @@ describe("withZod", () => {
       error: anyString,
     });
   });
+
+  it("returns custom error message when using a custom error map", async () => {
+    const schema = z.object({
+      type: z.string(),
+    });
+    const obj = {
+      type: "foo",
```

Comment:
This test is failing because this object is actually valid. Should be an easy fix, I think.

```suggestion
      type: 123,
```

Reply:
Oops my bad! Happy for you to fix this up and get it merged :D